accepted/00102-upgrade-to-salt-3008.md

```diff
@@ -0,0 +1,202 @@
+2. Do not package Salt Extensions at all -> builtin extensions + manual customization.
+3. Only package a reduced list of Salt Extensions -> we could consider some popular extensions that we should then maintain in the context of SLE.
+
+With option 1, we do package and do support all those extensions we decide to have a package for. On the other hand, with option 2, we do only take care of maintaing and supporting the reduced list of builin extensions (actually integrated in the main `python311-salt` package), and then users and customers would need to rely on extending with their desired extensions coming from the community.
+
+In case of packaging certain list of Salt Extensions, those will be available as RPM packages, allowing the ability to install those packaged ones in an system without access to internet or PyPI. Additionally, providing extensions via Salt Master file roots is also a valid option for disconnected setups.
+
+## Customizing your Salt package
+
+As described by Salt [here](https://salt.tips/the-great-salt-module-migration/), starting with Salt 3008, both Salt Master and Minion can be extended by using `salt-pip` or `pip.installed` to install the Salt Extensions packages (`saltext-*`). Similar to the extending mechanism we currently have for the Salt Bundle.
+
+If an extension is not existing (as most of the dropped modules are NOT yet migrated to Salt Extensions), they mention to provide them as custom modules in your Salt state tree (file_roots).
+
+As mentioned above, we would be probably providing a main Salt package containing some builtin extensions already there that at least covers basic operations.
+
+## Releasing the new Salt
+
+Salt 3008 won't support Python 3.6, so we cannot use the main Python 3.6 interpreter from SLE15 or Leap15, we need to make it to be based on Python 3.11.
+
+Besides of adapting the Salt specfile to build using Python 3.11, by using singlespec adaptations at https://github.com/openSUSE/salt-packaging/pull/96 (and related PRs), we need to take some actions when it comes to how this new version would be distributed in SLE15 and Leap15:
+
+#### SLE15:
+- A new custom "SLE-Module-Salt" module (inherited from SLE-Module-Python3) will be created for SLE15SP4/5/6/7.
+- The new Salt 3008.x package will be shipped via the new "SLE-Module-Salt" together with its any new dependency which is not part of the "SLE-Module-Python3" module.
+- The old Salt 3006.0 package in SLE15SP4/5/6 will be deprecated in favor of the new version coming in the new "SLE-Module-Salt".
+- The old Salt 3006.0 package will be dropped from SLE15SP7 Basesystem and Server Application modules before SLE15SP7 feature cutoff.
+
+#### Leap15:
+- Python 3.11 is already available in Leap.
+- The new Salt and any new dependencies will get into Leap from SLE.
+
+NOTE: SLMicro 6.0 works differently than SLE15, as it does contain Python 3.11 in the base channels.
```

Comment:
You should say also something about SLES12 SP5 LTSS and maybe about other OSes when it comes to SUMA.
AFAIK venv-salt-minion is the only affected package on these platforms as we do not have master and proxy there?

Reply:
The classic Salt package is already EOL in SLE12, as it was part of the "Advanced Systems Management Module" which does not have LTSS. The version there was actually Salt 3000, as 3006.0 was not even supported.

For RES8 and Ubuntu 20.04 (client tools), we still ship a classic Salt package (3006.0), I'll add some notes to the RFC also on this regard.

Reply:
I just added some notes to cover the cases for other OSes.